src/modules/job-manager/event.c

```diff
@@ -316,22 +340,37 @@ int event_job_post (struct event_ctx *ctx, struct job *job,
     return -1;
 }
 
-int event_job_post_fmt (struct event_ctx *ctx, struct job *job,
-                        flux_continuation_f cb, void *arg, const char *name,
-                        const char *fmt, ...)
+int event_job_post (struct event_ctx *ctx, struct job *job,
+                    flux_continuation_f cb, void *arg,
+                    const char *name, const char *context)
+{
+    return post (ctx, job, cb, arg, name, context);
+}
+
+int event_job_post_pack (struct event_ctx *ctx, struct job *job,
+                         flux_continuation_f cb, void *arg, const char *name,
+                         const char *fmt, ...)
 {
     va_list ap;
-    char context[FLUX_KVS_MAX_EVENT_CONTEXT + 1];
-    int n;
+    json_t *o = NULL;
+    char *context = NULL;
+    int ret = -1;
 
     va_start (ap, fmt);
-    n = vsnprintf (context, sizeof (context), fmt, ap);
-    va_end (ap);
-    if (n < 0 || n >= sizeof (context)) {
+    if (!(o = json_vpack_ex (NULL, 0, fmt, ap))) {
         errno = EINVAL;
-        return -1;
+        goto error;
     }
-    return event_job_post (ctx, job, cb, arg, name, context);
+    if (!(context = json_dumps (o, JSON_COMPACT))) {
+        errno = EINVAL;
+        goto error;
```

Comment:
I think you may get into trouble with unbalanced `va_start()` / `va_end()` here?